salt/roster/cache.py

```diff
@@ -162,16 +214,21 @@ def _data_lookup(ref, lookup):
 
 def _minion_lookup(minion_id, key, minion):
     grains, pillar, addrs, mine = minion
+
     if key == 'id':
         # Just paste in the minion ID
         return minion_id
+    elif key.startswith('sdb://'):
```

Comment:
you will want to make sure that key isinstance six.string_types here.  Otherwise, it will try to do a .startswith on a dictionary.